certbot/tests/compat/filesystem_test.py

```diff
@@ -0,0 +1,168 @@
+
+        self.assertFalse([dacl.GetAce(index) for index in range(0, dacl.GetAceCount())
+                          if dacl.GetAce(index)[2] == everybody])
+
+        filesystem.chmod(probe_path, 0o704)
+        dacl = _get_security_dacl(probe_path).GetSecurityDescriptorDacl()
+
+        self.assertTrue([dacl.GetAce(index) for index in range(0, dacl.GetAceCount())
+                         if dacl.GetAce(index)[2] == everybody])
+
+    def test_group_permissions_noop(self):
+        probe_path = _create_probe(self.tempdir)
+
+        filesystem.chmod(probe_path, 0o700)
+        ref_dacl_probe = _get_security_dacl(probe_path).GetSecurityDescriptorDacl()
+
+        filesystem.chmod(probe_path, 0o740)
+        cur_dacl_probe = _get_security_dacl(probe_path).GetSecurityDescriptorDacl()
+
+        self.assertTrue(filesystem._compare_dacls(ref_dacl_probe, cur_dacl_probe))  # pylint: disable=protected-access
+
+    def test_admin_permissions(self):
+        import win32security  # pylint: disable=import-error
+        probe_path = _create_probe(self.tempdir)
+
+        system = win32security.ConvertStringSidToSid('S-1-5-18')
+        admins = win32security.ConvertStringSidToSid('S-1-5-32-544')
+
+        filesystem.chmod(probe_path, 0o700)
+        dacl = _get_security_dacl(probe_path).GetSecurityDescriptorDacl()
+
+        self.assertTrue([dacl.GetAce(index) for index in range(0, dacl.GetAceCount())
+                         if dacl.GetAce(index)[2] == system])
+        self.assertTrue([dacl.GetAce(index) for index in range(0, dacl.GetAceCount())
+                         if dacl.GetAce(index)[2] == admins])
+
+    def test_read_flag(self):
+        import ntsecuritycon  # pylint: disable=import-error
+        self._test_flag(4, ntsecuritycon.FILE_GENERIC_READ)
+
+    def test_execute_flag(self):
+        import ntsecuritycon  # pylint: disable=import-error
+        self._test_flag(1, ntsecuritycon.FILE_GENERIC_EXECUTE)
+
+    def test_write_flag(self):
+        import ntsecuritycon  # pylint: disable=import-error
+        self._test_flag(2, (ntsecuritycon.FILE_ALL_ACCESS
+                            ^ ntsecuritycon.FILE_GENERIC_READ
+                            ^ ntsecuritycon.FILE_GENERIC_EXECUTE
+                            ^ 512))
+
+    def test_full_flag(self):
+        import ntsecuritycon  # pylint: disable=import-error
+        self._test_flag(7, (ntsecuritycon.FILE_ALL_ACCESS
+                            ^ 512))
+
+    def _test_flag(self, everyone_mode, windows_flag):
+        # Note that flag are tested again `everyone`, not `user`, because practically these unit
+        # tests are executed with admin privilege, so current user is effectively the admins group,
+        # and so will always have all rights.
+        import win32security  # pylint: disable=import-error
+        probe_path = _create_probe(self.tempdir)
+
+        filesystem.chmod(probe_path, 0o700 + everyone_mode)
```

Comment:
super nit: They are equivalent in this case, but I usually see `|` (or) being used for things like modes instead of `+`.

Reply:
Fixed.